### PR TITLE
(DOCSP-13343): Android SDK-native Aggregation Examples

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
### Jira

- https://jira.mongodb.org/browse/DOCSP-13343

### Staged Changes (Requires MongoDB Corp SSO)

- [Android SDK - Examples - Query MongoDB - Aggregation](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-13343/sdk/android/examples/mongodb-remote-access/#aggregate-documents-in-a-collection)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/android/examples/mongodb-remote-access.txt [hunk 1]
```diff
@@ -605,8 +553,277 @@ Running this snippet produces output resembling the following:
...
+
+Group Documents
+~~~~~~~~~~~~~~~
+
+You can use the :manual:`$group
+</reference/operator/aggregation/group/>` stage to aggregate summary
+data for one or more documents. MongoDB groups documents based
+on the expression defined in the ``_id`` field of the ``$group`` stage.
+You can reference a specific document field by prefixing the field name
+with a ``$``.
```
Comment (dacharyc) [inline source/sdk/android/examples/mongodb-remote-access.txt:626]: Question: my aggregation may be a bit rusty, but isn't the `$` when paired with a field how you refer to a field's _value_? I didn't think you need a `$` to refer to a field; only if you need to do something with the field's value?

Ahh, right, the next line refers to the field value, so maybe just clarify that here in this line?

File: source/sdk/android/examples/mongodb-remote-access.txt [hunk 2]
```diff
@@ -605,8 +553,277 @@ Running this snippet produces output resembling the following:
...
+
+The following snippet groups all documents in the ``plants``
+collection by their ``type`` value and aggregates a count of the
+number of each type:
+
+.. tabs-realm-languages::
+
+   .. tab::
```
Comment (dacharyc) [inline source/sdk/android/examples/mongodb-remote-access.txt:634]: Same as above re: the code examples - might make it slightly easier for readers to parse if you add a few comments breaking up what the code is doing, i.e. "Connect to the database..." etc.

File: source/sdk/android/examples/mongodb-remote-access.txt [hunk 3]
```diff
@@ -605,8 +553,277 @@ Running this snippet produces output resembling the following:
...
+         :language: java
+
+.. _android-mongodb-aggregation-project-document-fields:
+
+Project Document Fields
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the :manual:`$project
+</reference/operator/aggregation/project/>` stage to include or omit
+specific fields from documents or to calculate new fields using
```
Comment (dacharyc) [inline source/sdk/android/examples/mongodb-remote-access.txt:653]: Nit: bullets might be nice here, i.e.:
- Include specific fields in documents
- Omit specific fields from documents
- Calculate new fields using aggregation operators

If you opt for no bullets here, I might recommend a new sentence at "or to calculate..." - it's just a bit long and may be difficult for readers to parse.

File: source/sdk/android/examples/mongodb-remote-access.txt [hunk 4]
```diff
@@ -605,8 +553,277 @@ Running this snippet produces output resembling the following:
...
+:manual:`aggregation operators </reference/operator/aggregation>`.
+Projections work in two ways:
+
+- Explicitly include fields with a value of 1. This has the side-effect
```
Comment (dacharyc) [inline source/sdk/android/examples/mongodb-remote-access.txt:657]: Nit: I love the bullets here! But found the "explicitly include... implicitly exclude" stuff difficult to parse. I had to re-read a couple of times to understand what this was talking about. I wonder if we could clarify and simplify this? i.e.: 

> A `1` denotes fields that you want to include. If you use a `1` to specify any field to include, the projection omits any fields that are not listed with a `1`. Use this method when you want to include only a few fields, and omit most of the document.

Or something along those lines?

**Review by dacharyc — APPROVED** [review]
Nice work! It's great to see some more examples of aggregations in action, since they're such a powerful tool but can be difficult to grok. I have a handful of comments here, but they're largely just nits; feel free to take or leave them.

File: source/sdk/android/examples/mongodb-remote-access.txt [hunk 5]
```diff
@@ -605,8 +553,277 @@ Running this snippet produces output resembling the following:
...
+
+.. example::
+
+   The following ``$match`` stage filters documents to include
+   only those where the ``type`` field has a value equal to "perennial":
+
+   .. tabs-realm-languages::
+
+      .. tab::
+         :tabid: kotlin
```
[the remark this answers is not in the document]
Comment (nathan-contino-mongo) [reply source/sdk/android/examples/mongodb-remote-access.txt:605]: Hm, the biggest thing keeping me from doing this is simply that I don't want to make these examples much longer. But I do agree that they're a bit monolithic as-is. I'll see if I can clear them up a bit.

File: source/sdk/android/examples/mongodb-remote-access.txt (hunk 4 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/examples/mongodb-remote-access.txt:657]: Personally, I think I prefer the less verbose version that's already there, so I'm going to leave this as-is. IMO this is just a complicated concept that the reader is going to have to reread a couple of times to fully understand, and I'd rather keep it minimal than stretch it out over more sentences.

File: source/sdk/android/examples/mongodb-remote-access.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/android/examples/mongodb-remote-access.txt:606]: I like what you've done with these! They make the code snippets a lot easier to parse, I think. 👍

**Pull request merged**